mono/metadata/Makefile.am

```diff
@@ -433,6 +437,10 @@ libmonoruntimesgen_la_SOURCES = $(common_sources) $(icall_tables_sources) $(ilge
 libmonoruntimesgen_la_CFLAGS = $(SGEN_DEFINES) @CXX_ADD_CFLAGS@
 libmonoruntimesgen_la_LIBADD = libmonoruntime-config.la libmonoruntime-support.la
 
+libmonoruntimesgencore_la_SOURCES = $(libmonoruntimesgen_la_SOURCES)
+libmonoruntimesgencore_la_CFLAGS = $(libmonoruntimesgen_la_CFLAGS) -D NETCORE
```

Comment:
`-D NETCORE`

it should be `ENABLE_NETCORE`, and the define should be in `config.h`, that is, generated by configure.